upload/helpers.py

```diff
@@ -739,3 +751,42 @@ def validate_activated_repo(repository):
         raise ValidationError(
             f"This repository has been deactivated. To resume uploading to it, please activate the repository in the codecov UI: {settings_url}"
         )
+
+
+def get_agent_from_headers(headers):
+    try:
+        return headers["User-Agent"].split("/")[0].split("-")[1]
+    except Exception as e:
+        log.warning(
+            "Error getting agent from user agent header",
+            extra=dict(
+                err=str(e),
+            ),
+        )
+        return "unsupported-user-agent"
```

Comment:
[nit] maybe "unknown" is better than "unsupported" here IF (and only if, I actually don't know) we actually process the upload